src/cache/cache_las.c

```diff
@@ -826,10 +852,10 @@ __wt_las_cursor_position(WT_SESSION_IMPL *session, WT_CURSOR *cursor, uint32_t b
             return (0);
         else if (las_btree_id == btree_id) {
             WT_RET(__wt_compare(session, NULL, &las_key, key, &cmp));
-            if (cmp > 0)
+            if (cmp < 0)
                 return (0);
             else if (cmp == 0)
-                if (las_timestamp >= timestamp)
+                if (las_timestamp <= timestamp)
```

Comment:
There shouldn't be an `else` clause after a `return`.

I'd also suggest a compound `if (cmp == 0 && las_timestamp <= timestamp)`.

Reply:
Done.